transform-service/controller/src/main/java/org/apache/beam/transformservice/controller/Controller.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.transformservice.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.model.pipeline.v1.Endpoints;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.vendor.grpc.v1p54p0.io.grpc.Server;
+import org.apache.beam.vendor.grpc.v1p54p0.io.grpc.ServerBuilder;
+
+public class Controller {
+
+  List<Endpoints.ApiServiceDescriptor> endpoints;
+
+  private final TransformServiceOptions options;
+
+  public Controller(String[] args) {
+    // We use PipelineOptions just a library for parsing arguments here.
+    this(PipelineOptionsFactory.fromArgs(args).create());
+    for (String expansionService : options.getTransformServiceConfig().getExpansionservices()) {
```

Comment:
Should endpoints be initialized in `Controller(PipelineOptions opts)`, otherwise `Controller(String[] args)` has an initialized endpoints while `Controller(PipelineOptions opts)` has an empty endpoints. Both constructor are public.

also, any javadoc is nice for this public class and is executable.